src/ripple/rpc/impl/Utilities.cpp

```diff
@@ -94,5 +96,23 @@ injectSLE (Json::Value& jv,
     }
 }
 
+boost::optional<Json::Value> readLimitField(
+    unsigned int& limit,
+    Tuning::LimitRange const& range,
+    Context const& context)
+{
+    limit = range.rdefault;
+    if (auto const& jvLimit = context.params[jss::limit])
+    {
+        if (! (jvLimit.isUInt() || (jvLimit.isInt() && jvLimit.asInt() >= 0)))
+            return RPC::expected_field_error (jss::limit, "unsigned integer");
+
+        limit = jvLimit.asUInt();
+        if (context.role != Role::ADMIN)
+            limit = std::max(range.rmin, std::min(range.rmax, limit));
+    }
+    return boost::none;
+}
+
```

Comment:
Because `limit` is an output-only parameter, what do you think about this?
```
std::pair<boost::optional<Json::Value>, unsigned int>
readLimitField(
    Tuning::LimitRange const& range,
    Context const& context)
{
    auto limit = range.rdefault;
    if (auto const& jvLimit = context.params[jss::limit])
    {
        if (! (jvLimit.isUInt() || (jvLimit.isInt() && jvLimit.asInt() >= 0)))
            return { RPC::expected_field_error(jss::limit, "unsigned integer"),
                0 };

        limit = jvLimit.asUInt();
        if (context.role != Role::ADMIN)
            limit = std::max(range.rmin, std::min(range.rmax, limit));
    }
    return{ boost::none, limit };
}
```
The callers would then become:
```
    auto limitRes = readLimitField(RPC::Tuning::***whatever***, context);
    if (limitRes.first)
        return *limitRes.first;
    auto limit = limitRes.second;
```
The pair stays in scope if there's no error, which is unfortunate, but may be worth it.